### PR TITLE
Clarify XmlSerializingContext usage.

### PR DESCRIPTION
**Pull request by krocard**
The XmlSerializingContext usage was not very clear.

Add a doxygen for the class XmlSerializingContext and remove an unused
function that contradict with the recommended usage.

### CODE REVIEW

**Comment by krocard** [comment]
:+1: @dawagner @Makohoek

File: xmlserializer/XmlSerializingContext.h [hunk 1]
```diff
@@ -31,6 +31,15 @@
 
 #include <string>
 
+/** Class that gather errors during serialization.
+ *
+ * Provided with an empty buffer (strError), an error can be described.
+ *
+ * This error will be accessible formated in the buffer
+ * provided at construction (strError) *after* destruction.
```
Comment (dawagner) [inline xmlserializer/XmlSerializingContext.h]: "in the aforementioned buffer *after* destruction".

File: xmlserializer/XmlSerializingContext.h [hunk 2]
```diff
@@ -31,6 +31,15 @@
 
 #include <string>
 
+/** Class that gather errors during serialization.
+ *
+ * Provided with an empty buffer (strError), an error can be described.
```
Comment (dawagner) [inline xmlserializer/XmlSerializingContext.h]: Master Yoda, is that you ?

**Comment by dawagner** [comment]
:+1:

**Pull request merged**